src/main/java/org/tugraz/sysds/runtime/privacy/PrivacyConstraint.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.tugraz.sysds.runtime.privacy;
+
+/**
```

Comment:
While the design is fresh in memory, shall we add the implementation related documentation or design in this file.
Perhaps something like [this](https://github.com/apache/systemml/blob/cf74661016928e3413d693f939a67964f3256b19/src/main/java/org/apache/sysds/runtime/instructions/gpu/context/GPUMemoryManager.java#L62)